Source/WebKit/NetworkProcess/storage/IDBStorageRegistry.cpp

```diff
@@ -46,14 +46,18 @@ WebCore::IDBServer::IDBConnectionToClient& IDBStorageRegistry::ensureConnectionT
 
 void IDBStorageRegistry::removeConnectionToClient(IPC::Connection::UniqueID connection)
 {
-    auto allConnectionsToClient = std::exchange(m_connectionsToClient, { });
-    for (auto& [identifier, connectionToClient] : allConnectionsToClient) {
-        if (connectionToClient->ipcConnection() != connection) {
-            m_connectionsToClient.add(identifier, WTFMove(connectionToClient));
-            continue;
+    m_connectionsToClient.removeIf([&](auto& entry) {
+        auto& connectionToClient = entry.value;
+        if (!connectionToClient)
+            return true;
+
+        if (connectionToClient->ipcConnection() == connection) {
+            connectionToClient->connectionToClient().connectionToClientClosed();
```

Comment:
What if this delegate call ends up modifying `m_connectionsToClient`? Seems the old code was more robust in this regard.